src/test/java/com/thebois/models/inventory/InventoryTests.java

```diff
@@ -0,0 +1,108 @@
+package com.thebois.models.inventory;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.thebois.models.inventory.items.IItem;
+import com.thebois.models.inventory.items.ItemType;
+import com.thebois.models.inventory.items.Log;
+import com.thebois.models.inventory.items.Rock;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class InventoryTests {
+
+    public static Stream<Arguments> getItemTypes() {
+        return Stream.of(Arguments.of(ItemType.LOG), Arguments.of(ItemType.ROCK));
+    }
+
+    @ParameterizedTest
+    @MethodSource("getItemTypes")
+    public void emptyInventoryIsEmpty(final ItemType itemType) {
+        // Arrange
+        final AbstractInventory inventory = new ColonyInventory();
+
+        // Act
+        final Optional<IItem> result = inventory.takeItem(itemType);
+
+        // Assert
+        assertThat(result.isEmpty()).isTrue();
+    }
+
+    @Test
+    public void getsNoItemWhenSpecifiedTypeDoesNotExistEvenIfInventoryIsNotEmpty() {
```

Comment:
style

I'd reword it to `takeItemReturnsEmptyWhenOnlyOtherItemsInInventory`

Reply:
👍